internal/partitions/partition.go

```diff
@@ -94,66 +96,76 @@ func (p *Partition) lowerBoundFromCurrent(current bson.Raw) (any, error) {
-			// inconsistencies.
-			{"afterClusterTime", startAt},
-		}},
-		// The cursor should not have a timeout.
-		{"noCursorTimeout", true},
+type PartitionQueryParameters struct {
+	filter    option.Option[bson.D]
+	sortField option.Option[string]
+	hint      option.Option[bson.D]
+}
+
+func (pqp PartitionQueryParameters) ToFindOptions() bson.D {
+	doc := bson.D{}
+
+	if theFilter, has := pqp.filter.Get(); has {
+		doc = append(doc, bson.E{"filter", theFilter})
+	}
+
+	pqp.addHintIfNeeded(&doc)
+
+	return doc
+}
+
+func (pqp PartitionQueryParameters) ToAggOptions() bson.D {
+	pl := mongo.Pipeline{}
+
+	if theFilter, has := pqp.filter.Get(); has {
+		pl = append(pl, bson.D{{"$match", theFilter}})
 	}
-	if len(batchSize) > 0 {
-		findCmd = append(findCmd, bson.E{"batchSize", batchSize[0]})
+
+	if theSort, has := pqp.sortField.Get(); has {
+		pl = append(pl, bson.D{{"$sort", bson.D{{theSort, 1}}}})
 	}
-	findOptions := p.GetFindOptions(nil, nil)
-	findCmd = append(findCmd, findOptions...)
 
-	return findCmd
+	doc := bson.D{
+		{"pipeline", pl},
+	}
+
+	pqp.addHintIfNeeded(&doc)
+
+	return doc
+}
+
+func (pqp PartitionQueryParameters) addHintIfNeeded(docRef *bson.D) {
+	if theHint, has := pqp.hint.Get(); has {
+		*docRef = append(*docRef, bson.E{"hint", theHint})
+	}
 }
 
-// GetFindOptions returns only the options necessary to do a find on any given collection with this
-// partition. It is intended to allow the same partitioning to be used on different collections
-// (e.g. use the partitions on the source to read the destination for verification)
+// GetQueryParameters returns a PartitionQueryParameters that describes the
+// parameters needed to fetch docs for the partition. It is intended to allow
+// the same partitioning to be used on different collections (e.g. use the
+// partitions on the source to read the destination for verification)
 // If the passed-in buildinfo indicates a mongodb version < 5.0, type bracketing is not used.
 // filterAndPredicates is a slice of filter criteria that's used to construct the "filter" field in the find option.
-func (p *Partition) GetFindOptions(clusterInfo *util.ClusterInfo, filterAndPredicates bson.A) bson.D {
+func (p *Partition) GetQueryParameters(clusterInfo *util.ClusterInfo, filterAndPredicates bson.A) PartitionQueryParameters {
+	params := PartitionQueryParameters{}
+
 	if p == nil {
 		if len(filterAndPredicates) > 0 {
-			return bson.D{{"filter", bson.D{{"$and", filterAndPredicates}}}}
+			params.filter = option.Some(bson.D{{"$and", filterAndPredicates}})
+			return params
 		}
-		return bson.D{}
+
+		return params
 	}
-	findOptions := bson.D{}
+
 	if p.IsCapped {
 		// For capped collections, sort the documents by their natural order. We deliberately
 		// exclude the ID filter to ensure that documents are inserted in the correct order.
-		sort := bson.E{"sort", bson.D{{"$natural", 1}}}
-		findOptions = append(findOptions, sort)
+		params.sortField = option.Some("$natural")
```

Comment:
(nit) It feels weird to me to call this `sortField` when `$natural` is not a field.